src/coreclr/jit/optcse.cpp

```diff
@@ -3318,6 +3323,30 @@ class CSE_Heuristic
             bool doCSE = PromotionCheck(&candidate);
 
 #ifdef DEBUG
+
+            if (doCSE)
+            {
+                const int attempt = m_pCompiler->optCSEattempt++;
+
+                if (m_pCompiler->info.compMethodHash() == (unsigned)JitConfig.JitCSEHash())
+                {
+                    // We can only mask the first 32 CSE attempts, so suppress anything beyond that.
+                    // Note methods with > 32 CSEs are currently quite rare.
+                    //
+                    if (attempt > 32)
```

Comment:
```suggestion
                    if (attempt >= 32)
```

otherwise if `attempt == 32` you do (int) `1 << 32` which is undefined?